### ISSUE TITLE
Add mute button to scoreboard

### ISSUE THREAD

**Issue by FMSMITH91** [FMSMITH91/prophunt#5]

**Comment by gonzalezx600** [comment]
This should work I'm not sure if it still works tho I've used it last year and it worked so give it a shot!
If you want me to do it just say so and I will just cant do it at the moment as thats why I just put the link on what I did!

https://facepunch.com/showthread.php?t=1400474

**Comment by FMSMITH91** [comment]
Heads up — the mute button added here has never actually worked.

While checking whether the other closed issues had survived the upstream syncs (#4 turned out to have been silently reverted), I found that `canMute()` in `gamemodes/base_phx/gamemode/cl_scores.lua` calls:

```lua
LocalPlayer():CheckGroup(ply:GetUserGroup())
```

`Player:CheckGroup` is not a GMod method and is defined nowhere in this repo — the only occurrence in the entire codebase is that call site. So the scoreboard's mute column throws `attempt to call method 'CheckGroup' (a nil value)` for every other player it draws a row for.

Fix in #7: use the rule the F1 menu already applies in `prop_hunt/gamemode/cl_menutypes.lua`, where staff and usergroups whitelisted in `PHX.IgnoreMutedUserGroup` aren't mutable and everyone else is. That table is already synced to clients, so no new plumbing needed.

The button itself and the column wiring are still in place — only the permission check was broken.

**Comment by FMSMITH91** [comment]
**Correction to my previous comment — I was wrong.**

I claimed `Player:CheckGroup` doesn't exist. It does: it's a [ULib](https://github.com/TeamUlysses/ulib) method, defined on the player meta in [`lua/ulib/shared/sh_ucl.lua:300`](https://github.com/TeamUlysses/ulib/blob/master/lua/ulib/shared/sh_ucl.lua). From its own documentation:

> This function is similar to `IsUserGroup()`, but this one checks the UCL group chain as well. For example, if a user is in group "superadmin" which inherits from "admin", this function will return true if you check the user against "admin", where `IsUserGroup()` wouldn't.

So the original code was correct and intentional:

```lua
return LocalPlayer():CheckGroup( ply:GetUserGroup() )
```

That reads as *"is my rank at or above this player's?"* — an admin can mute a user, a user can't mute an admin. A deliberate permission model built on ULX roles, and it has worked fine on any server with ULib installed.

My error: I searched only this repository, found the sole occurrence was the call site, and concluded the method was undefined. I never checked whether a dependency provided it — and ULX/ULib add methods to the Player metatable, which is exactly the mechanism this gamemode's own `PHXIsStaff` uses. Grepping one repo isn't evidence a method doesn't exist when addons are in the picture.

The mute button was **not** broken. #7 changed its behaviour rather than fixing it.

#10 restores the ULib call, keeping only the parts of #7 that were genuinely right: `CheckGroup` really is nil on a server without ULib, so it's now guarded with a fallback to PH:X's own rule, and a `muted` variable that was leaking into `_G` on every scoreboard row stays scoped.

Apologies for the noise on a closed issue.
